### PR TITLE
chore(perf): prevent unnecessary bundling of caniuse-lite

### PR DESCRIPTION
**Pull request by st3iny**
## Summary

It does not support tree shaking as they are still building a cjs bundle. However, they bundle certain futures to separate smaller modules which we can take advantage of.

Saves us around 930 KB (prod bundle).

grafik

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]

**Review by susnux — APPROVED** [review]

**Review by ShGKme — APPROVED** [review]

**Review by joshtrichards — COMMENTED** [review]
Any downsides/trade-offs you can think of? Is this basically just including only what we need now and less of what we never used anyway?

**Comment by st3iny** [comment]
> Any downsides/trade-offs you can think of? Is this basically just including only what we need now and less of what we never used anyway?

The code that I removed from the bundle is not used at all so it's just an optimization.

File: core/src/views/UnsupportedBrowser.vue [hunk 1]
```diff
@@ -125,7 +125,7 @@ export default {
 	async beforeMount() {
 		// Dynamic load big list of user agents
 		// eslint-disable-next-line n/no-extraneous-import
-		const { agents } = await import('caniuse-lite')
+		const { agents } = await import('caniuse-lite/dist/unpacker/agents.js')
```
Comment (ShGKme) [inline core/src/views/UnsupportedBrowser.vue:128]: > // Dynamic load big list of user agents

Is it still actual? Do we need a dynamic load here?

Comment (st3iny) [reply core/src/views/UnsupportedBrowser.vue:128]: Nope not any more. I removed the dynamic import.

**Pull request merged**